pyccel/ast/core.py

```diff
@@ -201,78 +199,6 @@ def subs(expr, new_elements):
     else:
         return expr
 
-
-def allocatable_like(expr, verbose=False):
-    """
-    finds attributes of an expression
-
-    Parameters
-    ----------
-    expr: Expr
-        a pyccel expression
-
-    verbose: bool
-        talk more
-    """
-
-    if isinstance(expr, (Variable, IndexedElement)):
-        return expr
-    elif isinstance(expr, str):
-        # if the rhs is a string
-        return expr
-    elif isinstance(expr, Expr):
-        args = [expr]
-        while args:
-            a = args.pop()
-            # XXX: This is a hack to support non-Basic args
-            if isinstance(a, str):
-                continue
-
-            if a.is_Mul:
-                if _coeff_isneg(a):
-                    if a.args[0] is S.NegativeOne:
-                        a = a.as_two_terms()[1]
-                    else:
-                        a = -a
-                (n, d) = fraction(a)
-                if n.is_Integer:
-                    args.append(d)
-                    continue  # won't be -Mul but could be Add
-                elif d is not S.One:
-                    if not d.is_Integer:
-                        args.append(d)
-                    args.append(n)
-                    continue  # could be -Mul
-            elif a.is_Add:
-                aargs = list(a.args)
-                negs = 0
-                for ai in aargs:
-                    if _coeff_isneg(ai):
-                        negs += 1
-                        args.append(-ai)
-                    else:
-                        args.append(ai)
-                continue
-            if a.is_Pow and a.exp is S.NegativeOne:
-                args.append(a.base)  # won't be -Mul but could be Add
-                continue
-            if a.is_Mul or a.is_Pow or a.is_Function or \
-                    isinstance(a, (Derivative, Integral)):
-
-                o = Symbol(a.func.__name__.upper())
-            if not a.is_Symbol and not isinstance(a, (IndexedElement,
-                    FunctionCall)):
-                args.extend(a.args)
-
-            if isinstance(a, (Variable, IndexedElement)):
-                return a
-            elif a.is_Symbol:
-                raise TypeError('Found an unknown symbol {0}'.format(str(a)))
-    else:
-        raise TypeError('Unexpected type {0}'.format(type(expr)))
```

Comment:
This function uses lots of sympy objects which don't seem to be used elsewhere. Codacy often misses new `unused-import` errors which appear due to code removal. Could you run pylint locally to try and clean this up? The command is:
```
pylint ast/core.py --disable=all --enable=unused-variable
```

Reply:
For example `Derivative`

Reply:
(You can install pylint using pip)

Reply:
I'm getting 
```
Your code has been rated at 10.00/10
```

Reply:
Sorry, wrong command:
```
pylint ast/core.py --disable=all --enable=unused-import
```